python/ray/experimental/state/state_cli.py

```diff
@@ -308,12 +314,33 @@ def list(
     timeout: float,
     address: str,
 ):
-    """
-    List RESOURCE used by Ray.
+    """List all states of a given resource.
 
-    RESOURCE is the name of the possible resources from `StateResource`,
-    i.e. 'jobs', 'actors', 'nodes', ...
+    Normally, summary APIs are recommended before listing all resources.
 
+    The output schema is defined at :ref:`State API Schema section. <state-api-schema>`
+
+    For example, the output schema of `ray list tasks` is
+    :ref:`ray.experimental.state.common.TaskState <state-api-schema-task>`.
+
+    The API queries one or more components from the cluster to obtain the data.
+    The returned state snanpshot could be stale, and it is not guaranteed to return
+    the live data.
+
+    The API can return partial or missing output upon the following scenarios.
+
+    - When the API queries more than 1 component, if some of them fail,
+      the API will return the partial result (with a suppressable warning).
+    - When the API returns too many entries (10K), the API
```

Comment:
We probably want to embed constant values here or reference the exact consts or remember to change it once we figure out the fixed system upperbound for this.